Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md

```diff
@@ -0,0 +1,98 @@
+# Upgrading python interpreter version from 3.7.10 to 3.8.12
+
+Oppia changed the Python version it uses from 3.7.10 to 3.8.12 in Oct 2022. This page is aimed at developers who are currently using Python 3.7.10 and need to upgrade to 3.8.12. 
+
+For effective and effortless switching to the newer version, this guide will provide a brief and detailed instruction on how to achieve just that. Please follow the below mentioned instructions carefully.
+
+## Prerequisite
+
+This page is aimed at developers who want to upgrade their Python interpreter in their existing Oppia directory, instead of creating a separate copy of Oppia with the new interpreter.
+
+Before starting, please make sure that your local repo, as well as your remote fork on GitHub, are both up-to-date with the `develop` branch.
+
+## Step 1. Installing python 3.8.12 through pyenv
+Our first step is to install the new interpreter version `3.8.12` through pyenv by issuing the following command through a terminal :
+```
+pyenv install 3.8.12
+```
+After you execute the above command you can expect some output as depicted below :
+```
+❯ pyenv install 3.8.12
+Installing Python-3.8.12...
+patching file Misc/NEWS.d/next/Build/2021-10-11-16-27-38.bpo-45405.iSfdW5.rst
+patching file configure
+patching file configure.ac
+Installed Python-3.8.12 to /home/user/.pyenv/versions/3.8.12
+```
+
+## Step 2. Determining your virtualenv manager
+
+Oppia developers and contributors use either `pyenv` or `direnv` for creating a python virtualenv and automatic switching of python version.
+
+This step will guide you through determining which one you are using currently, so that you can update your virtualenv accordingly.
+
+#### Do the following :
+
+* Go to the directory where you have installed Oppia and check if you have `.envrc` file in that directory. If the `.envrc` file is present, that means you are using direnv for virtualenv activation. `direnv` users should follow the instructions in [Step 3 (a)](#step-3-(a).-setting-up-virtualenv-for-direnv-users).
```

Comment:
Please do not use \`xxx\` for files so that they are not confused with commands. Use _italics_ for example

Reply:
Done !!